TACC/Core-Styles
# feat(elements): root, demo & docs v. cms v. portal

## Overview

1. **Integrate https://github.com/TACC/TACC-Docs Core-Styles code.**
2. Distinguish root and table elements for different clients.
3. _Integrate a tiny piece of https://github.com/TACC/tup-ui/ code._

## Related

- integrates code from https://github.com/TACC/tup-ui/pull/174
    <sup>https://github.com/TACC/tup-ui/blob/task/save-css-snippets/apps/tup-cms/src/taccsite_cms/templates/snippets/css-elements-body.html#L2-L8</sup>

## Changes & Testing & UI

1. **monospace, shrink-wrap code block**
    http://localhost:3000/components/detail/monospace--docs
    http://localhost:3000/components/detail/monospace--cms
    1. Verify code blocks are shrink-wrapped.
    2. Verify scrolling code blocks retain background color for entire width.
    
    <details><summary>Screenshots</summary>

    | | scroll left | scroll right |
    | - | - | - |
    | docs | ![docs monospace scroll left](https://user-images.githubusercontent.com/62723358/224418467-4ad28b4f-0e06-46ce-88a1-7becce3fd6e6.png) | ![docs monospace scroll right](https://user-images.githubusercontent.com/62723358/224418471-ec7c15a3-8c0d-48fc-a18d-e4f4af489c16.png) |
    | cms | ![cms monospace scroll left](https://user-images.githubusercontent.com/62723358/224418474-f61a1f95-c41a-47b4-bca4-ddd797a4a1d8.png) | ![cms monospace scroll right](https://user-images.githubusercontent.com/62723358/224418475-0a760478-ea82-4413-9749-add1bf2e4759.png) |

    </details>
2. **headings--docs, do not color h2**
    http://localhost:3000/components/detail/headings--docs
    http://localhost:3000/components/detail/headings--cms
	1. Verify Docs `<h2>` is dark gray (not purple).
	2. Verify CMS `<h2>` is dark olive yellow (not dark gray).
	3. Verify CMS `<h2>` is dark gray (or white) in a section.
    
    <details><summary>Screenshots</summary>
    
    | cms (no section) | docs (no section) |
    | - | - |
    | ![cms headings](https://user-images.githubusercontent.com/62723358/224419177-bb4ddad9-69c0-4a19-8f41-44bfd40296dc.png) | ![docs headings no section](https://user-images.githubusercontent.com/62723358/224419171-29176fd8-b654-496c-a44d-937cb7e36a70.png) |

    | unstyled section | light section | muted section | dark section |
    | - | - | - | - |
    | ![cms headings unstyled section](https://user-images.githubusercontent.com/62723358/224419163-44b4f4a5-8246-40a2-a310-7652884c5e62.png) | ![cms headings light section](https://user-images.githubusercontent.com/62723358/224419158-e7881744-435c-4a7c-a430-28e5e17cc3a8.png) | ![cms headings muted section](https://user-images.githubusercontent.com/62723358/224419169-77b98184-bca3-45ea-987f-56c6542810e9.png) | ![cms headings dark section](https://user-images.githubusercontent.com/62723358/224419166-1f2a9352-c1b9-4420-9cc1-40cb52afeb8d.png)  |

    </details>
3. **docs table font size**
    http://localhost:3000/components/detail/table--docs
    http://localhost:3000/components/detail/table--cms
    - Verify Docs basic table has larger font than CMS basic table.
    
    <details><summary>Screenshots</summary>

    | docs | cms |
    | - | - |
    | ![docs table](https://user-images.githubusercontent.com/62723358/224420813-7e8249a6-f024-491e-8b3c-3730bb055f76.png) | ![cms table](https://user-images.githubusercontent.com/62723358/224420819-8c53a1a9-9fdb-4a13-bd49-013d310c0def.png) |

    </details>
4. **core-styles.base.css loads generics fonts**
    http://localhost:3000/components/detail/root--portal
    http://localhost:3000/components/detail/root--docs
    http://localhost:3000/components/detail/root--cms
    1. Verify Docs & CMS font is BentonSans **and** Portal font is Roboto.
        <sup>Do not just check CSS. Check Dev Tools: Elements: Computed (scroll to bottom): Rendered Fonts.</sup>
    
    <details><summary>Screenshots</summary>

    | portal | docs | cms |
    | - | - | - |
    | ![portal](https://user-images.githubusercontent.com/62723358/224421109-6978ccbf-b6fe-4b55-a4fc-962a14f758a0.png) | ![docs](https://user-images.githubusercontent.com/62723358/224421111-d45a628a-d3ea-4073-9944-44bbb1a79410.png) | ![cms](https://user-images.githubusercontent.com/62723358/224421112-01d29b3d-4b7d-453e-85ce-a71417f18516.png) |


    </details>
5. **pseudo-elements, scroll-margin-top**
    http://localhost:3000/components/preview/headings--cms#headings--dark-section
    - Verify clicking an anchor link does not bring content flush with top of page.
        <sup>There should be 20px of extra space. Noticeable in the example by a difference in background.</sup>
    
    <details><summary>Screenshots</summary>

    | cms dark section scroll margin |
    | - |
    | ![cms dark section scroll margin](https://user-images.githubusercontent.com/62723358/224421375-4a66aa88-1f42-4d73-b66e-a9dde853b9e1.png) |

    </details>
6. **table, use variables for cms**
    http://localhost:3000/components/detail/table--cms
    - Verify CMS basic table has these styles:
        - `font size` `14px`
        - `line-height` `1.4` i.e. `19.6px`
    
    <details><summary>Screenshots</summary>

    | cms basic table font size and line height.png |
    | - |
    | ![cms basic table font size and line height](https://user-images.githubusercontent.com/62723358/224421617-372502cb-be5b-4e8a-96db-33a6e959b9c5.png) |

    </details>
7. **root, demo & docs v. cms v. portal**
    http://localhost:3000/components/detail/root--default
    http://localhost:3000/components/detail/root--cms
    http://localhost:3000/components/detail/root--portal
    http://localhost:3000/components/detail/root--docs
    1. Verify CMS root page has `font-size` `14px`, `letter-spacing` `.025rem`, `line-height` `1.6` i.e. `22.4px`.
    2. Verify Portal root page has `font-size` `12px`, `line-height` `1.4` i.e. `16.8px`.
    3. Verify Docs root page has `font-size` `16px`, `line-height` `1.4` i.e. `22.4px`.
    
    <details><summary>Screenshots</summary>

    | default | cms | docs | portal |
    | - | - | - | - |
    | ![default root](https://user-images.githubusercontent.com/62723358/224421927-60ace986-48fd-46aa-8c01-e45149d2ed67.png) | ![cms root](https://user-images.githubusercontent.com/62723358/224421924-a8071337-0bb3-4a0e-8795-c50223e9160b.png) | ![docs root](https://user-images.githubusercontent.com/62723358/224421929-e7ed235b-7c02-4e1e-8dc5-da47e125dbd8.png) | ![portal root](https://user-images.githubusercontent.com/62723358/224421932-31b54081-0474-41e2-b0c6-cb999ad9d292.png) |

    </details>
8. **demo / default root styles**
    http://localhost:3000/components/detail/root--default
    http://localhost:3000/components/detail/x-link
    http://localhost:3000/components/detail/x-tabs
    http://localhost:3000/components/detail/s-affixed-input-wrapper
    http://localhost:3000/components/detail/s-blockquote
    http://localhost:3000/components/detail/s-irregular-links
    et cetera
    - Verify non-CMS-nor-Docs-nor-Portal page has `line-height` `1.4` i.e. `22.4px`, `font-family` any sans serif font.
    
    <details><summary>Screenshots</summary>

    | x-link | x-tabs | s-affixed-input-wrapper | s-blockquote | s-irregular-links |
    | - | - | - | - | - |
    | ![x link](https://user-images.githubusercontent.com/62723358/224422382-03c50eac-1525-4acb-9b93-ccf1a28d783d.png) | ![x tabs](https://user-images.githubusercontent.com/62723358/224422384-f8a89927-68ac-459f-9065-e99d498110c5.png) | ![s affixed input wrapper](https://user-images.githubusercontent.com/62723358/224422385-880ac344-e37e-4060-9ff3-7e3bbe129a3e.png) | ![s blockquote](https://user-images.githubusercontent.com/62723358/224422386-7a51ca28-9297-4518-a386-613a720e1a63.png) | ![links irregular](https://user-images.githubusercontent.com/62723358/224422389-12ab8239-b6e8-47ee-9142-b4164cb1b30d.png) |

    </details>
